src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
       }
     }
     if (errors) {
-      // TODO ERROR CORRECTION
-      throw new Error('Error correction not implemented yet');
+      // Perform B-M decoding
+      var syndromes = computeSyndromePoly(receivedMsg, n, k);
+      var bm = berlekampMassey(syndromes, n, k);
+      var errorLocations = chienSearch(bm.errorLocationPoly);
+      var errorMagnitudes = forneysFormula(errorLocations,
+                                           bm.errorMagnitudePoly);
```

Comment:
See below but I think this polynomial should be renamed for clarity. IE: I was confused by its name :P